jetty-security/src/main/java/org/eclipse/jetty/security/ConstraintMapping.java

```diff
@@ -18,17 +18,29 @@
 
 package org.eclipse.jetty.security;
 
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.http.pathmap.ServletPathSpec;
 import org.eclipse.jetty.util.security.Constraint;
 
 public class ConstraintMapping
```

Comment:
It feels like ConstraintMapping should be a subclass of Mapping.  It could extend Mapping, but override all the array-manipulating methods to ensure that the array only ever has 0/1 elements.

Reply:
I'll give it a try, but it's a little ugly